apps/elixir_ls_debugger/lib/debugger/server.ex

```diff
@@ -887,23 +867,26 @@ defmodule ElixirLS.Debugger.Server do
     end)
   end
 
-  defp ensure_thread_id(state = %__MODULE__{}, pid) when is_pid(pid) do
+  defp ensure_thread_id(state = %__MODULE__{}, pid, new_ids) when is_pid(pid) do
     if Map.has_key?(state.threads_inverse, pid) do
-      {state, state.threads_inverse[pid]}
+      {state, state.threads_inverse[pid], new_ids}
     else
       id = state.next_id
       state = put_in(state.threads[id], pid)
       state = put_in(state.threads_inverse[pid], id)
       state = put_in(state.next_id, id + 1)
-      {state, id}
+      {state, id, [id | new_ids]}
     end
```

Comment:
this might be a little cleaner using case
```elixir
case state.threads_inverse do 
  %{^pid => thread_id} -> {state, thread_id, new_ids}
   _ -> 
      ...
end
```

Reply:
I'm not convinced about readability improvement but it'll save one map lookup operation (unless the optimiser is smarter than I expect it to be)